h2o-core/src/main/java/water/rapids/ast/prims/models/AstFeatureImportance4BBM.java

```diff
@@ -0,0 +1,54 @@
+package water.rapids.ast.prims.models;
+
+
+import hex.Model;
+import water.Key;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.Env;
+import water.rapids.FeatureImportance4BBM;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.rapids.vals.ValFrame;
+
+
+import java.util.Map;
+import java.util.Map.Entry;
+
+import java.util.HashMap;
+
+public class AstFeatureImportance4BBM extends AstPrimitive {
+
+    @Override
+    public int nargs() {return 1 + 2;} // Perm_feature_importance + Frame + Model
+
+    @Override
+    public String[] args() {return new String [] {"frame", "model"};} 
+    
+    @Override
+    public String str() {return "Perm_Feature_importance";} // or call it model_reliance 
+    
+    @Override
+    public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]){
+        Frame fr = stk.track(asts[1].exec(env)).getFrame();
+        Model model = stk.track(asts[2].exec(env)).getModel();
+
+        Frame fr2 = model.score(fr); // prediction on original dataset
+        
+//      model._train.get(); TODO find a way to avoid passing the frame and get it from the model instead
```

Comment:
It would be nice to add the possibility of using the training dataset from the model. There is its id in model output. However, sometimes it could be useful to use different dataset so that I would keep both possibilities. For example, if the input dataset is null, use the training dataset from the model.

Reply:
hmm, it certainly would, good point.